B반 6조 배진영

jbae9 opened the pull request:
어렵지만 재밌고 문제 풀면서 많이 배웠습니다🥲

jbae9 committed:
테스트 통과 ^.ㅜ

hurricanenara commented on index.ts:
```diff
@@ -92,9 +97,15 @@ export function getSum(number1, number2) {}
 
-export function isStatusPending(status) {}
+export function isStatusPending(status: Status):boolean {
+    if(status === Status.Pending) return true
+    else return false
+}
 
-export function isStatusComplete(status) {}
+export function isStatusComplete(status: Status):boolean {
+    if(status === Status.Complete) return true
+    else return false
+}
```
한단계 더 간단하게 구현하실 수 있습니다 -- 확인바랍니다!

hurricanenara commented on index.ts:
```diff
@@ -131,17 +160,35 @@ export function getCars(): Car[] {}
 
 // 반환 타입
-type TProgrammingLanguages = "";
+type TProgrammingLanguages = {
+    [n: number]: keyof typeof ProgrammingLanguage
+}
+
```
`keyof typeof`가 필요한지 확인바랍니다

hurricanenara commented on index.ts:
```diff
@@ -131,17 +160,35 @@ export function getCars(): Car[] {}
 
-export function getProgrammingLanguages() {}
+export function getProgrammingLanguages():TProgrammingLanguages[] {
+    const answer: TProgrammingLanguages[] = Object.entries(ProgrammingLanguage).map(([str, programmingLanguage], index) => {
+        return {
+            [str.length]: programmingLanguage
+        }
+    })
```
`index`는 쓰이지 않는걸로 보입니다, 쓰이지 않는 코드는 제거해주세요

hurricanenara commented on index.ts:
```diff
@@ -131,17 +160,35 @@ export function getCars(): Car[] {}
 
-type TOrder = {};
+type TOrder = {
+    [key:string]: {
+        status: keyof typeof Status,
```
status 프로퍼티에 Status enum 자체만 반영해도 괜찮습니다

hurricanenara commented on index.ts:
```diff
@@ -131,17 +160,35 @@ export function getCars(): Car[] {}
+    [key:string]: {
+        status: keyof typeof Status,
+        color: number,
+        availableColors: [number],
```
`availableColors` 타입 확인 부탁드립니다

hurricanenara commented on index.ts:
```diff
@@ -167,9 +214,15 @@ const orders = {
+type TCustomerCar = {
+    [key:number]: {
+        customerLastName: Customer["lastName"],
+        car: number,
```
`car` 타입 확인 부탁드립니다

jbae9 committed:
- isStatusPending 과 isStatusComplete 함수들 더 간략하게 수정
- getProgrammingLanguages 함수에 쓰이지 않는 매개변수 "_" 처리
- TOrder 타입에 status property 수정
